Source/SessionDelegate.swift

```diff
@@ -49,12 +49,14 @@ open class SessionDelegate: NSObject {
             assertionFailure("StateProvider is nil.")
             return nil
         }
+//
+//        guard let request = provider.request(for: task) as? R else {
+//            fatalError("Returned Request is not of expected type: \(R.self).")
+//        }
+//
+//        return request
 
-        guard let request = provider.request(for: task) as? R else {
-            fatalError("Returned Request is not of expected type: \(R.self).")
-        }
-
-        return request
+        return provider.request(for: task) as? R
```

Comment:
Is this a straight replacement? If yes, can we remove the commented out code?

Reply:
Yep, needed to be able to check for multiple types. Removed in https://github.com/Alamofire/Alamofire/pull/3074/commits/a12ab983b06efeccf77d0ec86d02e85f4a99b92a.